### PR TITLE
feat(pipeline): add SAM2 image segmentation prototype

### PR DESCRIPTION
**Pull request by rickstaa**
This pull request introduces a prototype implementation of the [Segment Anything v2](https://github.com/facebookresearch/segment-anything-2) (SAM2) pipeline within the AI worker. The prototype demonstrates the basic functionality needed to perform segmentation on an image. Note that video segmentation is not yet implemented. Additionally, the dependencies were updated quickly, which may temporarily break other pipelines.

### CODE REVIEW

**Comment by rickstaa** [comment]
@pschroedl,

Here is a quick prototype for the [Segment Anything v2 pipeline](https://github.com/facebookresearch/segment-anything-2.git). I decided to begin with image segmentation first, as there are still many design choices we need to make to integrate this into the Livepeer network.

> **IMPORTANT**
> To expedite the process, I updated the dependencies to those required by the model. This likely broke other pipelines or at least the `sfast` optimization. This highlights the importance of the more customized pipeline structure you are developing.

File: runner/app/routes/segment_anything_2.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+        return JSONResponse(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content=http_error("SAM2 error"),
+        )
+
+    return {
+        "masks": json.dumps(masks.tolist()),
+        "iou_predictions": json.dumps(iou_predictions.tolist()),
+        "low_res_masks": json.dumps(low_res_masks.tolist()),
```
Comment (rickstaa) [inline runner/app/routes/segment_anything_2.py:109]: @pschroedl, what do you think about adding a boolean parameter like `low_res` to control the quality of the mask returned? This could help reduce data transfer when high resolution isn't necessary. Then if set we return the low_res_mask instead of the masks 🤔?

**Comment by rickstaa** [comment]
@pschroedl closing this for https://github.com/livepeer/ai-worker/pull/165.

File: runner/app/routes/segment_anything_2.py (hunk 1 above)
Comment (pschroedl) [reply runner/app/routes/segment_anything_2.py]: As requested by users, we're going to return all three by default, but renaming [predictions -> scores, mask -> logits](https://github.com/livepeer/ai-worker/pull/154/commits/b200ab6aaf98310855d4e7013452b3b3a2f49b08) to follow expected convention
